src/Cli.Tests/ValidateConfigTests.cs

```diff
@@ -60,4 +60,20 @@ public void TestConfigWithInvalidConfigProperties()
         bool isConfigValid = ConfigGenerator.IsConfigValid(validateOptions, _runtimeConfigLoader!, _fileSystem!);
         Assert.IsFalse(isConfigValid);
     }
+
+    /// <summary>
+    /// This method validates that the IsConfigValid method returns false when the config is empty.
+    /// This is to test no exception gets thrown with validate during failure config deserialization.
```

Comment:
how are you testing that no exception is thrown? By ensuring `ConfigGenerator.IsConfigValid(...)` only returns a bool?

Reply:
in `IsConfigValid` there is a call to `runtimeConfigValidator.TryValidateConfig(...)` which calls `ValidateEntitiesMetadata()`. Could  `ValidateEntitiesMetadata()` throw an exception which resuls in a bool not being returned for `IsConfigValid`? I suppose this PR's change only affects the mismatched dataSourceName, but unclear if it prevents other exceptions from being printed to the console

Reply:
correct, this PR is just to fix the issue of dataSourceName not found. 
Since, I moved the deserialization logic to a single place, I added this test to make sure exceptions due to deserialization failures are captured correctly.